game/yodesk.ksy

```diff
@@ -0,0 +1,479 @@
+meta:
+  id: yodesk
+  application: "Star Wars: Yoda Stories"
+  file-extension: dta
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: le
+  encoding: ASCII
+seq:
+  - id: catalog
+    type: catalog_entry
+    repeat: until
+    repeat-until: _.type == "ENDF"
+types:
+  catalog_entry:
+    seq:
+      - id: type
+        type: str
+        size: 4
+      - id: content
+        type:
+          switch-on: type
+          cases:
+             '"VERS"': version
+             '"STUP"': setup_image
+             '"CHAR"': characters
+             '"CAUX"': character_auxiliaries
+             '"CHWP"': character_weapons
+             '"PUZ2"': puzzles
+             '"SNDS"': sounds
+             '"TILE"': tiles
+             '"TNAM"': tile_names
+             '"ZONE"': zones
+             '"ENDF"': endf
+             _: unknown_case_error
+  unknown_case_error:
+    seq:
+      - id: message
+        contents: "Value did not match any expected type"
+  version:
+    seq:
+      - id: version
+        type: u4
+  setup_image:
+    seq:
+      - id: size
+        type: u4
+      - id: pixels
+        size: size
+  sounds:
+    seq:
+      - id: size
+        type: u4
+      - id: count
+        type: s2
+      - id: sounds
+        type: prefixed_strz
+        repeat: expr
+        repeat-expr: -count
+  tile_names:
+    seq:
+      - id: size
+        type: u4
+      - id: names
+        type: tile_name
+        repeat: until
+        repeat-until:  _.tile_id == -1
+  tile_name:
+    seq:
+      - id: tile_id
+        type: s2
+      - id: name
+        type: str
+        size: 0x18
+        if: tile_id != -1
+  tiles:
+    seq:
+      - id: size
+        type: u4
+      - id: tiles
+        type: tiles_entries
+        size: size
+  tiles_entries:
+    seq:
+      - id: tiles
+        type: tile
+        repeat: eos
+  tile:
+      seq:
+        - id: attributes
+          type: u4
+        - id: pixels
+          size: 32 * 32
+
+  action:
+    seq:
+      - id: marker
+        contents: "IACT"
+      - id: size
+        type: u4
+      - id: condition_count
+        type: u2
+      - id: conditions
+        type: action_item
+        repeat: expr
+        repeat-expr: condition_count
+      - id: instruction_count
+        type: u2
+      - id: instructions
+        type: action_item
+        repeat: expr
+        repeat-expr: instruction_count
+  action_item:
+    seq:
+      - id: opcode
+        type: u2
```

Comment:
Are opcodes known? Should it be a enum?

Reply:
I split `action_item` into `instruction` and `condition` and added enums for the respective opcodes.